src/runtime/internal.ts

```diff
@@ -164,140 +169,82 @@ export type DetectLocaleContext = {
-      path,
-      strategy,
-      alwaysRedirect,
-      redirectOn,
-      locale
-    )
+  __DEBUG__ && logger.log({ locale, path, strategy, alwaysRedirect, redirectOn })
 
   if (strategy !== 'no_prefix') {
-    if (redirectOn === 'root') {
-      if (path !== '/') {
-        __DEBUG__ && console.log('detectBrowserLanguage: not root')
-        return { locale: '', stat: false, reason: 'not_redirect_on_root' }
-      }
-    } else if (redirectOn === 'no prefix') {
-      __DEBUG__ && console.log('detectBrowserLanguage: no prefix (path) -', path)
-      if (!alwaysRedirect && path.match(getLocalesRegex(localeCodes))) {
-        return { locale: '', stat: false, reason: 'not_redirect_on_no_prefix' }
-      }
+    // detection only on root
+    if (redirectOn === 'root' && path !== '/') {
+      __DEBUG__ && logger.log('not root', { path })
+      return { locale: '', reason: DetectFailure.NO_REDIRECT_ROOT }
+    }
+
+    __DEBUG__ && redirectOn === 'no prefix' && logger.log('no prefix -', { path })
+
+    // detection only on unprefixed route
+    if (redirectOn === 'no prefix' && !alwaysRedirect && path.match(getLocalesRegex(localeCodes))) {
+      return { locale: '', reason: DetectFailure.NO_REDIRECT_NO_PREFIX }
     }
   }
 
-  let localeFrom: DetectBrowserLanguageFrom = 'unknown'
-  let cookieLocale: string | undefined
-  let matchedLocale: string | undefined
+  // track detection match source
+  let from: DetectFrom | undefined
 
-  // get preferred language from cookie if present and enabled
+  // match locale from cookie if enabled and present
+  const cookieMatch = (useCookie && localeCookie) || undefined
   if (useCookie) {
-    matchedLocale = cookieLocale = localeCookie
-    localeFrom = 'cookie'
-    __DEBUG__ && console.log('detectBrowserLanguage: cookieLocale', cookieLocale)
+    from = DetectFrom.COOKIE
   }
-  // try to get locale from either navigator or header detection
-  if (!matchedLocale) {
-    matchedLocale = getBrowserLocale()
-    localeFrom = 'navigator_or_header'
-    __DEBUG__ && console.log('detectBrowserLanguage: browserLocale', matchedLocale)
-  }
-  __DEBUG__ &&
-    console.log(
-      'detectBrowserLanguage: (matchedLocale, cookieLocale, localeFrom) -',
-      matchedLocale,
-      cookieLocale,
-      localeFrom
-    )
-
-  // set fallback locale if that is not matched locale
-  const finalLocale = matchedLocale || fallbackLocale
-  if (!matchedLocale && fallbackLocale) {
-    localeFrom = 'fallback'
-  }
-  __DEBUG__ &&
-    console.log(
-      'detectBrowserLanguage: first finaleLocale (finaleLocale, cookieLocale, localeFrom) -',
-      finalLocale,
-      cookieLocale,
-      localeFrom
-    )
-
-  const vueI18nLocale = locale || vueI18nOptionsLocale
-  __DEBUG__ && console.log('detectBrowserLanguage: vueI18nLocale', vueI18nLocale)
-
-  // handle cookie option to prevent multiple redirects
-  if (finalLocale && (!useCookie || alwaysRedirect || !cookieLocale)) {
```

Comment:
Besides debug logs in these if statements, these all return the same values, and so could be simplified and removed